jenkins/bootstrap.py

```diff
@@ -569,7 +576,16 @@ def node():
     # TODO(fejta): jenkins sets the node name and our infra expect this value.
     # TODO(fejta): Consider doing something different here.
     if NODE_ENV not in os.environ:
-        os.environ[NODE_ENV] = ''.join(socket.gethostname().split('.')[:1])
+        host = socket.gethostname().split('.')[0]
+        try:
+            # Try reading the name of the VM we're running on, using the
+            # metadata server.
+            os.environ[NODE_ENV] = urllib2.urlopen(urllib2.Request(
```

Comment:
discussed offline: this is sorta hacky but then again so is bootstrap. /shrug
we might want to solve this in some other fashion in the podutils, still discussing..

Reply:
can we expose node name via k8s downward api in prow?

Reply:
That would work too, but I don't want to add 10000 lines to config.yaml. Is there a better way?

Reply:
not sure if podutil can decorate this cc @stevekuznetsov